client/src/main/java/org/eclipse/hono/client/CommandResponse.java

```diff
@@ -34,22 +33,24 @@
     private static final Predicate<Integer> INVALID_STATUS_CODE = code ->
         code == null || code < 200 || (code >= 300 && code < 400) || code >= 600;
 
-    private final Buffer payload;
-    private final String contentType;
-    private final int status;
+    private final Message message;
     private final String replyToId;
-    private final String correlationId;
-    private final Map<String, Object> properties = new HashMap<>();
 
     private CommandResponse(final String tenantId, final String deviceId, final Buffer payload,
             final String contentType, final int status, final String correlationId, final String replyToId) {
-        this.payload = payload;
-        this.contentType = contentType;
-        this.status = status;
+        message = ProtonHelper.message();
+        message.setCorrelationId(correlationId);
+        MessageHelper.setCreationTime(message);
+        MessageHelper.addProperty(message, MessageHelper.APP_PROPERTY_TENANT_ID, tenantId);
+        MessageHelper.addProperty(message, MessageHelper.APP_PROPERTY_DEVICE_ID, deviceId);
+        MessageHelper.addProperty(message, MessageHelper.APP_PROPERTY_STATUS, status);
+        MessageHelper.setPayload(this.message, contentType, payload);
+        this.replyToId = replyToId;
```

Comment:
why do we need to keep this property? Can't we determine the message's address from the replyToId and the other parameters?

Reply:
@sophokles73 Below method in `CommandConnection` takes _replyId_ as an argument.
`Future<CommandResponseSender> getCommandResponseSender(String tenantId, String replyId);` 

`AbstractProtocolAdapterBase` uses the above method to get a sender and it needs the _replyId_.

Reply:
I see :+1: